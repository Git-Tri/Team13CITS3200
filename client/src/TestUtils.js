const trueErrorLog = console.error;

/**
 * okay this is a weird one to explain
 * basically there an issue with our dev setup (web-pack/react/create react app)
 * where if you test when a component fails it will spam the error log with
 * a message telling you the component failed 
 * since we are testing that, this is completely unneccessary
 * as such i am filtering the console.error to exclude these messages 
 * to give a more readable test output
 * using monkey patching: https://davidwalsh.name/monkey-patching
 */
export function filterConsoleError()
{

    
    console.error = (e) => 
    {

        
        if(typeof(e) === "string" && ! e.includes("React will try to recreate this component tree from scratch using the error boundary you provided"))
        {

            trueErrorLog(e);
        
        }
        else if(typeof(e) !== "string")
        {

            trueErrorLog(e);

        }
        
    }

}

export function unfilterConsoleError()
{

    console.error = trueErrorLog;

}