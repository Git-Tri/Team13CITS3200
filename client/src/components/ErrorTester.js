import React from "react";

/**
 * Catches an error and uses the onError callback to allow tests to pass
 * this component is meant for testing purposes only
 * when testing place the error component as a child to this component 
 * then pass in an onError prop or check the state of this component
 * derived from: https://codepen.io/gaearon/pen/wqvxGa?editors=0010
 */
class ErrorTester extends React.Component {
    constructor(props) {
      super(props);
      this.state = { error: null, errorInfo: null };
    }
    
    componentDidCatch(error, errorInfo) {
      this.setState({
        error: error,
        errorInfo: errorInfo
      })

      this.props.onError(error,errorInfo);
    }
    
    render() {
      if (this.state.errorInfo) {
        return (
          <p> Good news an error occurred</p>
          );
      }
      return this.props.children;
    }  
  }

  export default ErrorTester;
