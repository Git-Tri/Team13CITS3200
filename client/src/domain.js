/**
 * A class representing unstructured data
 * for list you may want to leave the data field undefined
 * NOTE: this code is duplicated on client and server so is only tested on the server
 */
export class UnstructuredData
{

    constructor(id,matchid,title,author,url,published,extracted,data)
    {

        this.id = id;
        this.matchid = matchid;
        this.title = title;
        this.author = author;
        this.url = url;
        this.published = published;
        this.extracted = extracted;
        this.data = data; 

    }

}

/**
 * A class representing structured data, for lists you may want to leave data undefined
 */
export class StructuredData
{

    constructor(id,date,home,away,competitionID,competitionName,data)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;
        this.data = data; 

    }

}

/**
 * a class representing a match 
 */
export class Match
{

    constructor(id,date,home,away,competitionID,competitionName)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;

    }

}

class InsertMatch
{

    constructor(id,competitionID,date,home,away,data)
    {

        this.id = id;
        this.competitionID = competitionID;
        this.date = date;
        this.home = home;
        this.away = away;
        this.data = data;
          

    }

}

/**
 * A class representing an edit 
 */
export class Edit 
{

    constructor(editID,structuredDataID,unstructuredDataID,isCorpus,settings,replace,replaceWith,type,order)
    {

        this.editID = editID;
        this.structuredDataID = structuredDataID;
        this.unstructuredDataID = unstructuredDataID;
        this.isCorpus = isCorpus;
        this.settings = settings;
        this.replace = replace;
        this.replaceWith = replaceWith;
        this.type = type;
        this.order = order;
        
    }

}


export class Competition
{

    constructor(id,name,countryName,countryId)
    {

        this.id = id; 
        this.name = name; 
        this.countryName = countryName;
        this.countryId = countryId

    }

}

export class ImportRequest
{

    constructor(begin,end,compId)
    {

        this.begin = begin; //optional

        this.end = end; //optional

        this.compId = compId; //mandatory 

    }

}


export class SearchRequest
{

    constructor(type,value,field)
    {

        this.type = type;
        this.value = value;
        this.field = field; 

    }

}


export class User
{

    constructor(id, username, hash, admin, regkey, token, apikey)
    {

        this.id = id; 
        this.username = username; 
        this.hash = hash;
        this.admin = admin;
        this.regkey = regkey;
        this.token = token;
        this.apikey = apikey;

    }

}