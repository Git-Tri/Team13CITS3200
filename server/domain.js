/**
 * A class representing unstructured data
 * for list you may want to leave the data field undefined
 * NOTE: this code is duplicated on client and server so is only tested on the server
 */
class UnstructuredData
{

    constructor(id,matchid,title,author,url,published,extracted,data)
    {

        this.id = id;
        this.matchid = matchid;
        this.title = title;
        this.author = author;
        this.url = url;
        this.published = published;
        this.extracted = extracted;
        this.data = data; 

    }

}

/**
 * A class representing structured data, for lists you may want to leave data undefined
 */
class StructuredData
{

    constructor(id,date,home,away,competitionID,competitionName,data)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;
        this.data = data; 

    }

}

/**
 * a class representing a match 
 */
class Match
{

    constructor(id,date,home,away,competitionID,competitionName)
    {

        this.id = id;
        this.date = date;
        this.home = home;
        this.away = away;
        this.competitionID = competitionID;
        this.competitionName = competitionName;

    }

}

class InsertMatch
{

    constructor(id,competitionID,date,home,away,data)
    {

        this.id = id;
        this.competitionID = competitionID;
        this.date = date;
        this.home = home;
        this.away = away;
        this.data = data;
          

    }

}

/**
 * A class representing an edit 
 */
class Edit 
{

    constructor(editID,structuredDataID,unstructuredDataID,isCorpus,settings,replace,replaceWith,type)
    {

        this.editID = editID;
        this.structuredDataID = structuredDataID;
        this.unstructuredDataID = unstructuredDataID;
        this.isCorpus = isCorpus;
        this.settings = settings;
        this.replace = replace;
        this.replaceWith = replaceWith;
        this.type = type;
        
    }

}

class Competition
{

    constructor(id,name,countryName,countryId)
    {

        this.id = id; 
        this.name = name; 
        this.countryName = countryName;
        this.countryId = countryId

    }

}

class User
{

    constructor(id, username, hash, admin, regkey)
    {

        this.id = id; 
        this.username = username; 
        this.hash = hash;
        this.admin = admin;
        this.regkey = regkey;

    }

}


class ImportRequest
{

    constructor(begin,end,compId)
    {

        this.begin = begin; //optional

        this.end = end; //optional

        if(typeof(compId !== "string"))
        {

            throw new Error("Competition must exist and of type string");

        }

        this.compId = compId; //mandatory 

    }

}

class SearchRequest
{

    constructor(type,value,field)
    {

        this.type = type;
        this.value = value;
        this.field = field; 

    }

}

//export all classes 
module.exports = 
{

    UnstructuredData,StructuredData,Match,Edit,Competition,ImportRequest, InsertMatch, SearchRequest, User

}