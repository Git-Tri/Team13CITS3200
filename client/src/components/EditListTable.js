import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import { Edit} from "../domain";

/**
 * shows a table of structured data 
 * has the ability to select a row 
 */
class EditListTable extends Component
{

    /**
     * constructs a new structured data table
     * data is mandatory however onSelect is optional 
     * @param {*} props the props passed in
     */
    constructor(props)
    {

        super(props);

        let selectFunc;

        if(this.props.onSelect != undefined)
        {
            selectFunc = (data) => 
                {
                    
                    this.setState({activeRow: data.id})

                    this.props.onSelect(data);

                }
            

        }
        else
        {

            selectFunc = () => {};

        }

        this.state = {activeRow: -1,selectFunc: selectFunc};

    }

    /**
     * renders the function
     * also checks if the data is valid 
     * only checks the fields which are relevant to this component 
     */
    render()
    {

        if(! Array.isArray( this.props.items))
        {

            throw Error("Props.items should contain a list of edits")

        }
        if(this.props.items.every((i) => i != undefined && i != null) == false)
        {

            throw new Error("props.data must be an instance of edit")

        }
        //borken into each predicate for easier debugging
        if(this.props.items.every((i) => 
        {
            let isEditID = Number.isInteger(i.editID);
            let isTargetValid = i.isCorpus || Number.isInteger(i.structuredDataID) || Number.isInteger(i.unstructuredDataID)
            let isTypeValid = typeof(i.type) == "string";
            let isSettingsValid = (typeof(i.settings) == "object" && i.type != "replacewithfield") || 
                (i.type == "replacewithfield" && typeof(i.settings) == "object" && i.settings !== undefined && i.settings !== null && typeof(i.settings.field) == "string");    
            let isReplaceValid = typeof(i.replace) == "string";
            let isReplaceWithValid = typeof(i.replaceWith) == "string";
            return isEditID && isTargetValid && isTypeValid && isSettingsValid && isReplaceValid && isReplaceWithValid;
        }) === false)
        {

            throw new Error("every piece of data should be a valid edit");

        }

        let isSelectable = this.props.onSelect !== undefined;

        let rows = this.props.items.map((item, index) => (
            <EditRow 
                key={item.editID} 
                data={item} 
                onSelect={this.state.selectFunc} 
                isActive={item.id == this.state.activeRow}
            />
        ))

        return(
        <Table striped selectable={isSelectable} >        
            <Table.Header>
                <Table.Row>
                    <Table.HeaderCell>Target</Table.HeaderCell>
                    <Table.HeaderCell>Type</Table.HeaderCell>
                    <Table.HeaderCell>Summary</Table.HeaderCell>
                </Table.Row>
            </Table.Header>  
            <Table.Body>
                {rows}
            </Table.Body>        
        </Table>
        );
    }
}

/**
 * builds a summary for an edit using it's type and other fields
 * @param {*} edit the edit to build a summary for 
 */
function SummaryBuilder(edit)
{


    switch(edit.type)
    {

        case "replace":
                return <div><i>Replace: </i> {edit.replace}  <i> - with - </i>  {edit.replaceWith}</div>

        case "replacewithfield":
            return <div><i>Replace: </i> {edit.replace}  <i> - with - </i>  {edit.replaceWith} 
                <i> - on field - </i>  {edit.settings.field} </div>

        default:
            return "Error";

    }


}

/**
 * formats the type field of an edit to something more human readable
 * @param {*} edit the edit to return the formatted type field to
 */
function FormatType(edit)
{

    switch(edit.type)
    {

        case "replace":
            return "Replace"

        case "replacewithfield":
            return "Replace with Field"

        default:
            return "Error";

    }

}

/**
 * Represents a row of the edit list table
 * as this component isn't exported no additional validation is done 
 * by this component 
 * @param {*} props 
 */
function EditRow(props)
{

    if((props.data instanceof Edit) == false)
    {

        throw Error("props.data must be an instance of structured data");

    }
    

    let edit = props.data;


    let sid = edit.structuredDataID;
    let uid = edit.unstructuredDataID;
    let isCorpus = edit.isCorpus;

    let target; 

    if(Number.parseInt(sid) > 0)
    {

        target = "Structured Data: " + sid; 

    }
    else if(Number.parseInt(uid) > 0)
    {

        target = "Unstructured Data: " + uid;

    }
    else if(isCorpus)
    {

        target = "All Corpus";

    }
    else
    {

        target = "Invalid";

    }



    let selectFunc = props.onSelect != undefined ? props.onSelect : () => {};

    return(
        <Table.Row onClick={() => selectFunc(props.data)} active={props.isActive}>
            <Table.Cell>{target}</Table.Cell>
            <Table.Cell>{FormatType(edit)}</Table.Cell>
            <Table.Cell>{SummaryBuilder(edit)}</Table.Cell>
        </Table.Row>
    );
    
}

export default EditListTable;