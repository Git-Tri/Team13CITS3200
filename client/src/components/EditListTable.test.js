import EditListTable from "./EditListTable";
import renderer from 'react-test-renderer';
import React from 'react';
import ErrorTester from "./ErrorTester";
import {Edit} from "../domain";
import {filterConsoleError,unfilterConsoleError} from "../TestUtils";
import {shallow, mount} from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe("Edit List Table Tests",function()
{

    beforeAll(() =>
        {

            return new Promise(function(resolve,reject)
            {

                filterConsoleError();

                resolve();

            })

        });   
    
    const testItems = [new Edit(1,1,null,false,{},"apple","banna","replace"),
        new Edit(2,null,1,false,{},"football","soccer","replace"),
        new Edit(3,1,null,false,{field:"player"},"gary jones","player 1","replacewithfield"),
        new Edit(4,null,null,true,{},"reece","ryan","replace")]    

    test("Should throw error when given undefined items",(done) => 
    {

    renderer.create(<ErrorTester              
             onError={(e,em) => {
                    expect(e.message).toEqual("Props.items should contain a list of edits"); 
                    done();}
                    }>
                 <EditListTable ></EditListTable>
             </ErrorTester>
            );       

    });

    test("Should throw error when given non-array items",(done) => 
    {

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("Props.items should contain a list of edits"); 
                   done();}
                   }>
                <EditListTable items={"not an item"}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error when one item is undefined",(done) => 
    {

        let missingItemData = testItems.slice();

        missingItemData.push(undefined);

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("props.data must be an instance of edit"); 
                   done();}
                   }>
                <EditListTable items={missingItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error when item id wrong type",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit("not an id",1,null,false,{},"apple","banna","replace"));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error with no target for edit",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit(132,null,null,false,{},"apple","banna","replace"));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error with wrong type",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit(12,1,null,false,{},"apple","banna",null));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error with missing settings for replacewithfield type",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit(21,1,null,false,{},"apple","banna","replacewithfield"));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error with wrong replace field",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit(1,1,null,false,null,undefined,"with","replace"));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should throw error with wrong replace with field",(done) => 
    {

        let wrongIDItemData = testItems.slice();

        wrongIDItemData.push(new Edit(1,1,null,false,null,"replace",undefined,"replace"));

        renderer.create(<ErrorTester              
            onError={(e,em) => {
                   expect(e.message).toEqual("every piece of data should be a valid edit"); 
                   done();}
                   }>
                <EditListTable items={wrongIDItemData}></EditListTable>
            </ErrorTester>
           );       

    });

    test("Should render correctly for correct input",() => 
    {        
                 
        let Component = renderer.create(<EditListTable items={testItems.slice()}/>)

        expect(Component.toJSON()).toMatchSnapshot();

    })

    test("Should render correctly for correct input with onSelect handler",() => 
    {        
        
        let Component = renderer.create(<EditListTable onSelect={() => {}} items={testItems.slice()}/>)

        expect(Component.toJSON()).toMatchSnapshot();

    })

    test("Should call callback when onselect is trigerred",(done) => 
    {

        let wrapper = mount(<EditListTable onSelect={() => done()} items={testItems.slice()}/>)

        wrapper.state().selectFunc(testItems[0]);


    })

    test("Should render correctly with selected row",() => 
    {

        let wrapper = shallow(<EditListTable onSelect={() => {}} items={testItems.slice()}/>)

        wrapper.state().selectFunc(testItems[0]);

        expect(wrapper).toMatchSnapshot();


    })

    test("Should render correctly for correct with no data",() => 
    {        
        
        let Component = renderer.create(<EditListTable onSelect={() => {}} items={[]}/>)

        expect(Component.toJSON()).toMatchSnapshot();

    })

    afterAll(() =>
    {

        return new Promise(function(resolve,reject)
        {

            unfilterConsoleError();

            resolve();

        })

    });

})

