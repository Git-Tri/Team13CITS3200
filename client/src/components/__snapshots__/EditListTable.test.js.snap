// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit List Table Tests Should render correctly for correct input 1`] = `
<table
  className="ui striped table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Target
      </th>
      <th
        className=""
      >
        Type
      </th>
      <th
        className=""
      >
        Summary
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        6/3/1991 - some comp : some team vs some other team
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          apple
            
          <i>
             - with - 
          </i>
            
          banna
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        7/4/1991 - some author : some title
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          football
            
          <i>
             - with - 
          </i>
            
          soccer
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        6/3/1991 - some comp : some team vs some other team
      </td>
      <td
        className=""
      >
        Replace with Field
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          gary jones
            
          <i>
             - with - 
          </i>
            
          player 1
          <i>
             - on field - 
          </i>
            
          player
           
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        No Data Selected
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          reece
            
          <i>
             - with - 
          </i>
            
          ryan
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Edit List Table Tests Should render correctly for correct input with onSelect handler 1`] = `
<table
  className="ui selectable striped table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Target
      </th>
      <th
        className=""
      >
        Type
      </th>
      <th
        className=""
      >
        Summary
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        6/3/1991 - some comp : some team vs some other team
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          apple
            
          <i>
             - with - 
          </i>
            
          banna
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        7/4/1991 - some author : some title
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          football
            
          <i>
             - with - 
          </i>
            
          soccer
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        6/3/1991 - some comp : some team vs some other team
      </td>
      <td
        className=""
      >
        Replace with Field
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          gary jones
            
          <i>
             - with - 
          </i>
            
          player 1
          <i>
             - on field - 
          </i>
            
          player
           
        </div>
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        No Data Selected
      </td>
      <td
        className=""
      >
        Replace
      </td>
      <td
        className=""
      >
        <div>
          <i>
            Replace: 
          </i>
           
          reece
            
          <i>
             - with - 
          </i>
            
          ryan
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Edit List Table Tests Should render correctly for correct with no data 1`] = `
<table
  className="ui selectable striped table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Target
      </th>
      <th
        className=""
      >
        Type
      </th>
      <th
        className=""
      >
        Summary
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  />
</table>
`;

exports[`Edit List Table Tests Should render correctly with selected row 1`] = `
<Table
  as="table"
  selectable={true}
  striped={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
      >
        Target
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Type
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Summary
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <EditRow
      data={
        Object {
          "edit": Edit {
            "editID": 1,
            "isCorpus": false,
            "replace": "apple",
            "replaceWith": "banna",
            "settings": Object {},
            "structuredDataID": 1,
            "type": "replace",
            "unstructuredDataID": null,
          },
          "target": StructuredData {
            "away": "some other team",
            "competitionID": 1,
            "competitionName": "some comp",
            "data": Object {},
            "date": 1991-04-20T00:00:00.000Z,
            "home": "some team",
            "id": 1,
          },
        }
      }
      isActive={true}
      key="1"
      onSelect={[Function]}
    />
    <EditRow
      data={
        Object {
          "edit": Edit {
            "editID": 2,
            "isCorpus": false,
            "replace": "football",
            "replaceWith": "soccer",
            "settings": Object {},
            "structuredDataID": null,
            "type": "replace",
            "unstructuredDataID": 1,
          },
          "target": UnstructuredData {
            "author": "some author",
            "data": "some data",
            "extracted": 1991-04-20T00:00:00.000Z,
            "id": 1,
            "matchid": 1,
            "published": 1991-04-20T00:00:00.000Z,
            "title": "some title",
            "url": "some url",
          },
        }
      }
      isActive={true}
      key="2"
      onSelect={[Function]}
    />
    <EditRow
      data={
        Object {
          "edit": Edit {
            "editID": 3,
            "isCorpus": false,
            "replace": "gary jones",
            "replaceWith": "player 1",
            "settings": Object {
              "field": "player",
            },
            "structuredDataID": 1,
            "type": "replacewithfield",
            "unstructuredDataID": null,
          },
          "target": StructuredData {
            "away": "some other team",
            "competitionID": 1,
            "competitionName": "some comp",
            "data": Object {},
            "date": 1991-04-20T00:00:00.000Z,
            "home": "some team",
            "id": 1,
          },
        }
      }
      isActive={true}
      key="3"
      onSelect={[Function]}
    />
    <EditRow
      data={
        Object {
          "edit": Edit {
            "editID": 4,
            "isCorpus": true,
            "replace": "reece",
            "replaceWith": "ryan",
            "settings": Object {},
            "structuredDataID": null,
            "type": "replace",
            "unstructuredDataID": null,
          },
          "target": undefined,
        }
      }
      isActive={true}
      key="4"
      onSelect={[Function]}
    />
  </TableBody>
</Table>
`;
